pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -246,6 +249,7 @@ cdef class _Timestamp(datetime):
             result.nanosecond = self.nanosecond
         return result
 
+    @cython.final
```

Comment:
can you perf check this (you can just compare time-its for couple of ops in PR and master to see); if it makes a diff then add an asv (and if not let's not do this).

Reply:
>  if it makes a diff then add an asv

I'm shocked to see there isn't already an asv for "Timestamp addition/subtraction".  I'll make one, or see if one isn't hidden somewhere other than benchmarks/timestamp.py.  If it is somewhere else, will moving them around cause a problem for backward-comparability?

> can you perf check this

Not seeing much difference, but %timeit results are jumping around more than I'd like.  I've got a couple of questions queued up for the cython mailing list, will add this to them.

Reply:
> I'm shocked to see there isn't already an asv for "Timestamp addition/subtraction". I'll make one, or see if one isn't hidden somewhere other than benchmarks/timestamp.py. If it is somewhere else, will moving them around cause a problem for backward-comparability?

no there are, its just asv's seem  to be unstable for you :> (and this is prob a tiny difference)

Reply:
> no there are

Unless the instability extends to the contents of the benchmarks/ files, I'm pretty sure asv_bench/benchmarks/timestamp.py doesn't have any addition/subtraction benchmarks.

Reply:
timeseries.py has plenty

Reply:
Yah, so the question about moving was if its OK to consolidate Timestamp asvs (kinda analogous to what I've been doing for the tests)

Reply:
yes sure. we definitly want compreshensive benchmarks for scalar + scalar (and scalar + index, etc). so consolidation is fine.